### PR TITLE
Feature/15 view patients list

### PR DESCRIPTION
**Pull request by harryul**

### CODE REVIEW

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs [hunk 1]
```diff
@@ -1,16 +1,59 @@
...
+            };
+            PatientList result = _patientService.GetListPatientPaging(request);
+            List<PatientViewModel> patientsVm = new List<PatientViewModel>();
+            foreach (Patient p in result.Patients)
+            {
+                patientsVm.Add(new PatientViewModel()
+                {
+                    PatientIdentifier = p.PatientIdentifier,
+                    FullName = p.FullName,
+                    Gender = p.Gender == 0 ? "Male" : p.Gender == 1 ? "Female" : "Other",
```
Comment (vietvo361) [inline src/PatientChecking/PatientChecking/Controllers/PatientController.cs]: Should create `PatientGender enum`.

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs [hunk 2]
```diff
@@ -1,16 +1,59 @@
...
+        {
+            PagingRequest request = new PagingRequest()
+            {
+                PageIndex = currentPage,
+                PageSize = pagingOption,
+                SortSelection = sortOption
+            };
+            PatientList result = _patientService.GetListPatientPaging(request);
+            List<PatientViewModel> patientsVm = new List<PatientViewModel>();
+            foreach (Patient p in result.Patients)
```
Comment (vietvo361) [inline src/PatientChecking/PatientChecking/Controllers/PatientController.cs:37]: Should add empty line above for readability. Please check for other places.

File: src/PatientChecking/PatientChecking/Views/Patient/Index.cshtml [hunk 3]
```diff
@@ -1,5 +1,163 @@
...
+        @foreach (var item in Model.Patients)
+        {
+            <div class="row">
+                <div class="col l-4">
+                    <div class="item patient-item1">
+                        <div class="patient-img-box">
+                            <img src="~/Image/no-avatar.jpg" alt="Avatar" style="height: 135px" />
+                        </div>
+                        <div class="patient-name">
+                            <p style="font-weight: 600; font-size: 18px">@item.FullName</p>
```
Comment (vietvo361) [inline src/PatientChecking/PatientChecking/Views/Patient/Index.cshtml]: Should move this  style to a css class, please don't use inline style. Please check for other places.

File: src/PatientChecking/PatientChecking/Views/ViewModels/IPagination.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientChecking.Views.ViewModels
+{
+    interface IPagination
+    {
+        public int PageIndex { get; set; }
```
Comment (vietvo361) [inline src/PatientChecking/PatientChecking/Views/ViewModels/IPagination.cs]: Should remove the public in interface

File: src/PatientChecking/PatientChecking/wwwroot/css/PatientList.css [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+.patient-item3 {
+    display: flex;
+    flex-direction: column;
+    height: 100%;
+    justify-content: space-evenly;
+}
+
+    .patient-item3 > div {
```
Comment (vietvo361) [inline src/PatientChecking/PatientChecking/wwwroot/css/PatientList.css]: Please adjust indent of this block

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs (hunk 1 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Controllers/PatientController.cs]: PatientGender has been created and applied. Please review it for me again.

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs (hunk 2 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Controllers/PatientController.cs:37]: empty line has been added. Please review this for me again.

File: src/PatientChecking/PatientChecking/Views/Patient/Index.cshtml (hunk 3 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Views/Patient/Index.cshtml]: Inline style has been all removed in code. Please review this for me again.

File: src/PatientChecking/PatientChecking/Views/ViewModels/IPagination.cs (hunk 4 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Views/ViewModels/IPagination.cs]: The public access modifier has been removed in interface.

File: src/PatientChecking/PatientChecking/wwwroot/css/PatientList.css (hunk 5 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/wwwroot/css/PatientList.css]: I believe this indent of block shouble be keeped for readability. But I will apply it in the next feature

File: src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs [hunk 6]
```diff
@@ -8,19 +10,31 @@ namespace PatientChecking.Services.ServiceModels
...
     {
         public Patient()
+        {
+            
+        }
+        public int PatientId { get; set; }
+        public string PatientIdentifier { get; set; } 
+        public string FullName { get; set; }
+        public DateTime DoB { get; set; }
+        public int Gender { get; set; }
```
Comment (vietvo361) [inline src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs]: Please use `PatientGender` enum

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs [hunk 7]
```diff
@@ -1,16 +1,64 @@
...
+
+            List<PatientViewModel> patientsVm = new List<PatientViewModel>();
+
+            foreach (Patient p in result.Patients)
+            {
+                patientsVm.Add(new PatientViewModel()
+                {
+                    PatientIdentifier = p.PatientIdentifier,
+                    FullName = p.FullName,
+                    Gender = p.Gender == 0 ? PatientGender.Male.ToString() : p.Gender == 1 ? PatientGender.Female.ToString() : PatientGender.Other.ToString(),
```
Comment (vietvo361) [inline src/PatientChecking/PatientChecking/Controllers/PatientController.cs]: If we use `PatientGender `enum for Patient class, just `p.Gender.ToString()` here

File: src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs (hunk 6 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs]: DoB's datatype has been changed to PatientGender enum

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs (hunk 7 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Controllers/PatientController.cs]: This has been applied in code.

File: src/PatientChecking/PatientChecking/Services/ServiceModels/Enum/PatientGender.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientChecking.Services.ServiceModels.Enum
+{
+    public enum PatientGender
+    {
+        Male,
+        Female,
+        Other
```
Comment (minhphien) [inline src/PatientChecking/PatientChecking/Services/ServiceModels/Enum/PatientGender.cs]: We dont use Other for Patient, instead, please use "None" to handle null

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs [hunk 9]
```diff
@@ -1,16 +1,64 @@
...
+
+        public PatientController(IPatientService patientService)
         {
-            return View();
+            _patientService = patientService;
+        }
+
+        public IActionResult Index(int sortOption = (int)PatientSortSelection.ID, int pagingOption = 10, int currentPage = 1)
+        {
+            PagingRequest request = new PagingRequest()
```
Comment (minhphien) [inline src/PatientChecking/PatientChecking/Controllers/PatientController.cs]: Use implicit typing for local variables when the type of the variable is obvious from the right side of the assignment, or when the precise type is not important.

Comment (minhphien) [reply src/PatientChecking/PatientChecking/Controllers/PatientController.cs]: which mean just use var request = new PagingRequest for local variable. please apply for all your code added.

File: src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs [hunk 10]
```diff
@@ -8,19 +11,31 @@ namespace PatientChecking.Services.ServiceModels
...
+        public DateTime DoB { get; set; }
+        public PatientGender Gender { get; set; }
+        public string AvatarLink { get; set; }
+        [NotMapped]
+        public Address PrimaryAddress { get; set; }
+        [NotMapped]
+        public Contact PrimaryContact { get; set; }
+    }
+
+    public class PatientDetails : Patient
```
Comment (minhphien) [inline src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs]: Move class to separate file please

File: src/PatientChecking/PatientChecking/Views/Shared/_Layout.cshtml [hunk 11]
```diff
@@ -30,7 +30,7 @@
                         Patients
                         <span class="fas fa-caret-down first" ></span>
                     </a>
-                    <ul class="@(@ViewBag.Controller == "Patient" ? "patient-show show" : "patient-show")" >
+                    <ul class="@(@ViewBag.Controller == "Patient" ? "patient-show show" : "patient-show")" style="padding-left:0px">
```
Comment (minhphien) [inline src/PatientChecking/PatientChecking/Views/Shared/_Layout.cshtml]: inline css was told you in last reviews but I see it still apply here.
If you follow Bootstrap docs,
https://getbootstrap.com/docs/5.0/utilities/spacing/
using style="padding-left:0px" is similar to adding class "ps-0".

**Review by minhphien — CHANGES_REQUESTED** [review]

File: src/PatientChecking/PatientChecking/Services/ServiceModels/Enum/PatientGender.cs (hunk 8 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Services/ServiceModels/Enum/PatientGender.cs]: PatientGender has been fixed and apply value for each item in enum

File: src/PatientChecking/PatientChecking/Controllers/PatientController.cs (hunk 9 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Controllers/PatientController.cs]: This has been applied as per reviewer request.

File: src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs (hunk 10 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Services/ServiceModels/Patient.cs]: I have moved all classes into separated files

File: src/PatientChecking/PatientChecking/Views/Shared/_Layout.cshtml (hunk 11 above)
Comment (harryul) [reply src/PatientChecking/PatientChecking/Views/Shared/_Layout.cshtml]: I will fix this in the new branch based on develop and fix layout as per reviewer request

Comment (harryul) [reply src/PatientChecking/PatientChecking/Views/Shared/_Layout.cshtml]: I have replaced in-line style with bootstrap class

**Review by minhphien — APPROVED** [review]

**Pull request merged**